pkgs/by-name/wi/winbox4/package.nix

```diff
@@ -1,128 +1,44 @@
-
-  installPhase = ''
-    runHook preInstall
-
-    install -Dm644 "assets/img/winbox.png" "$out/share/pixmaps/winbox.png"
-    install -Dm755 "WinBox" "$out/bin/WinBox"
-
-    wrapProgram "$out/bin/WinBox" --run "${lib.getExe finalAttrs.migrationScript}"
-
-    runHook postInstall
-  '';
-
-  desktopItems = [
-    (makeDesktopItem {
-      name = "winbox";
-      desktopName = "Winbox";
-      comment = "GUI administration for Mikrotik RouterOS";
-      exec = "WinBox";
-      icon = "winbox";
-      categories = [ "Utility" ];
-    })
-  ];
-
-  migrationScript = writeShellApplication {
-    name = "winbox-migrate";
-    text = ''
-      XDG_DATA_HOME=''${XDG_DATA_HOME:-$HOME/.local/share}
-      targetFile="$XDG_DATA_HOME/MikroTik/WinBox/Addresses.cdb"
-
-      if [ -f "$targetFile" ]; then
-        echo "NixOS: WinBox 4 data already present at $(dirname "$targetFile"). Skipping automatic migration."
-        exit 0
-      fi
-
-      # cover both wine prefix variants
-      # latter was used until https://github.com/NixOS/nixpkgs/pull/329626 was merged on 2024/07/24
-      winePrefixes=(
-        "''${WINEPREFIX:-$HOME/.wine}"
-        "''${WINBOX_HOME:-$XDG_DATA_HOME/winbox}/wine"
-      )
-      sourceFilePathSuffix="drive_c/users/$USER/AppData/Roaming/Mikrotik/Winbox/Addresses.cdb"
-      selectedSourceFile=""
-
-      for prefix in "''${winePrefixes[@]}"
-      do
-        echo "NixOS: Probing WinBox 3 data path at $prefix..."
-        if [ -f "$prefix/$sourceFilePathSuffix" ]; then
-          selectedSourceFile="$prefix/$sourceFilePathSuffix"
-          break
-        fi
-      done
-
-      if [ -z "$selectedSourceFile" ]; then
-        echo "NixOS: WinBox 3 data not found. Skipping automatic migration."
-        exit 0
-      fi
-
-      echo "NixOS: Automatically migrating WinBox 3 data..."
-      install -Dvm644 "$selectedSourceFile" "$targetFile"
-    '';
-  };
-
-  meta = {
+  metaCommon = {
     description = "Graphical configuration utility for RouterOS-based devices";
     homepage = "https://mikrotik.com";
     downloadPage = "https://mikrotik.com/download";
-    changelog = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/CHANGELOG";
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
     license = lib.licenses.unfree;
-    platforms = [ "x86_64-linux" ];
     mainProgram = "WinBox";
     maintainers = with lib.maintainers; [
       Scrumplex
       yrd
       savalet
     ];
   };
+  x86_64-zip = callPackage ./build-from-zip.nix {
+    inherit pname version metaCommon;
+
+    hash = "sha256-RZpsKew3BaId6+tcwUV6fniUpCH4wIP9ab6P5oE7OAk=";
+  };
+
+  x86_64-dmg = callPackage ./build-from-dmg.nix {
+    inherit pname version metaCommon;
+
+    hash = "sha256-ZxvuEVx9BmFukPMEPKeqXQNW38ExSpnRcSuHdw6j+CI=";
+  };
+in
+(if stdenvNoCC.hostPlatform.isDarwin then x86_64-dmg else x86_64-zip).overrideAttrs (oldAttrs: {
+  passthru = (oldAttrs.passthru or { }) // {
+    inherit x86_64-zip x86_64-dmg;
```

Comment:
Does that means that no matter which platform this is going to be, both platforms will be evaluated?

Reply:
> Does that means that no matter which platform this is going to be, both platforms will be evaluated?

The fail is evaluated but not the inner derivation

Reply:
It's a good question, I guess I will let someone else review this PR.

Reply:
But why do we need the packages in passthru here?